subprojects/build-cache-http/src/main/java/org/gradle/caching/http/internal/DefaultHttpBuildCacheServiceFactory.java

```diff
@@ -106,11 +107,13 @@ private HttpRedirectVerifier createRedirectVerifier(URI url, boolean allowInsecu
             .create(
                 url,
                 allowInsecureProtocol,
-                () -> DeprecationLogger.deprecate("Using insecure protocols with remote build cache, without explicit opt-in,")
-                    .withAdvice("Switch remote build cache to a secure protocol (like HTTPS) or allow insecure protocols.")
-                    .willBeRemovedInGradle7()
-                    .withDslReference(HttpBuildCache.class, "allowInsecureProtocol")
-                    .nagUser(),
+                () -> {
+                    String message =
+                        "Using insecure protocols with remote build cache, without explicit opt-in, is unsupported. " +
+                            "Switch remote build cache to a secure protocol (like HTTPS) or allow insecure protocols. " +
+                            Documentation.dslReference(HttpBuildCache.class, "allowInsecureProtocol");
+                    throw new InvalidUserCodeException(message);
```

Comment:
I don't see why using a local variable here helps. It does show a warning in IntelliJ about inlining said variable. I think that's good advice.

Reply:
Mostly for formatting and readability.